### PR TITLE
[SYCLomatic] Support migration of asm vadd4/vsub4/vabsdiff4/vmin4/vmax4.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/runtime/dpct-rt/include/math.hpp [hunk 1]
```diff
@@ -712,6 +712,25 @@ template <typename T> sycl::vec<int32_t, 2> extractAndExtend2(T a) {
   return ret;
 }
 
+template <typename T> sycl::vec<int32_t, 4> extractAndExtend4(T a) {
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/math.hpp]: Should we extend each half word to T?

Comment (tangjj11) [reply clang/runtime/dpct-rt/include/math.hpp]: [CUDA doc](https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#simd-video-instructions) extended them to signed 17-bit, so we use int32.

**Review by yihwang-nv — APPROVED** [review]
LGTM

File: clang/runtime/dpct-rt/include/dpct/math.hpp [hunk 2]
```diff
@@ -712,6 +712,25 @@ template <typename T> sycl::vec<int32_t, 2> extractAndExtend2(T a) {
   return ret;
 }
 
+template <typename T> sycl::vec<int32_t, 4> extractAndExtend4(T a) {
+  sycl::vec<int32_t, 4> ret;
+  sycl::vec<T, 1> va{a};
+  if constexpr (std::is_signed_v<T>) {
+    auto v = va.template as<sycl::vec<int8_t, 4>>();
+    ret[0] = zero_or_signed_extent(v[0], 17);
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/math.hpp]: comment1: v[0] is int8, so Line 720, we only 9 bit is enough.
comment2:  zero_or_signed_extent() return int64, should here using a zero_or_signed_extent() which return a 32int.

Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/math.hpp]: fixed

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**